src/cpp/src/visual_language/pipeline.cpp

```diff
@@ -179,7 +182,12 @@ class ov::genai::VLMPipeline::VLMPipelineImpl {
         auto end_get_inputs_embeds = std::chrono::steady_clock::now();
 
         auto to_remove_from_hist = m_inputs_embedder->get_num_tokens_to_remove_from_hist();
-        ov::genai::utils::trim_kv_cache(m_language, to_remove_from_hist, m_kv_cache_seq_length_axis, std::nullopt);
+        if (m_inputs_embedder->should_reset_kv_cache())
+            m_language.reset_state();
```

Comment:
why do we need special handling?

Reply:
I think we need to create issue on plugin(s)

Reply:
I'd expect `reset_state()` to work faster

Reply:
> I'd expect `reset_state()` to work faster

depends on the implementation. That issue should be fixed anyway.